arduino-ide-extension/src/browser/contributions/board-selection.ts

```diff
@@ -79,22 +80,83 @@ export class BoardSelection extends SketchContribution {
           );
           return;
         }
-        const boardDetails = await this.boardsService.getBoardDetails({
-          fqbn: selectedBoard.fqbn,
-        });
-        if (boardDetails) {
-          const { VID, PID } = boardDetails;
-          const detail = `BN: ${selectedBoard.name}
-VID: ${VID}
-PID: ${PID}`;
-          await remote.dialog.showMessageBox(remote.getCurrentWindow(), {
-            message: nls.localize('arduino/board/boardInfo', 'Board Info'),
-            title: nls.localize('arduino/board/boardInfo', 'Board Info'),
-            type: 'info',
-            detail,
-            buttons: [nls.localize('vscode/issueMainService/ok', 'OK')],
-          });
+        // IDE2 must show the board info based on the selected port.
+        // https://github.com/arduino/arduino-ide/issues/1489
+        // IDE 1.x does not support network ports
+        if (selectedPort.protocol === 'network') {
+          this.messageService.info(
+            nls.localize(
+              'arduino/board/noNetworkPort',
+              "Network port, can't obtain info."
+            )
+          );
+          return;
+        }
```

Comment:
I'm thinking this should be generalized to `if (selectedPort.protocol !== 'serial')`, with the notification text changed to something like "Non-serial port, can't obtain info".

We must keep in mind that ports can now have any arbitrary protocol instead of it only being `serial` and `network` as was the case for Arduino IDE 2.x. If the "Board Info" system is not able to correctly handle ports of arbitrary protocols, then they should be explicitly disabled.

For example, if I connect a Teensy board and select the port from the "**teensy ports**" section of the **Tools > Port** menu, when I select **Tools > Get Board Info**, Arduino IDE shows the incorrect notification:

> **Native serial port, can't obtain info.**